osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/SetEncryptionKeyExchangeOnGMeterRequestMessageProcessor.java

```diff
@@ -34,11 +38,8 @@ public SetEncryptionKeyExchangeOnGMeterRequestMessageProcessor() {
 
     @Override
     protected Serializable handleMessage(final DlmsConnectionManager conn, final DlmsDevice device,
-            final Serializable requestObject) throws OsgpException {
-
-        this.assertRequestObjectType(GMeterInfoDto.class, requestObject);
-
-        final GMeterInfoDto gMeterInfo = (GMeterInfoDto) requestObject;
-        return this.configurationService.setEncryptionKeyExchangeOnGMeter(conn, device, gMeterInfo);
+            final RequestWithMetadata<GMeterInfoDto> request) throws OsgpException {
+        return this.configurationService.setEncryptionKeyExchangeOnGMeter(conn, device,
+                CorrelatedObject.from(request, request.getRequestObject()));
```

Comment:
A call to `CorrelatedObject.from(request)` seems nicer, as it takes both metadata and object from `RequestWithMetadata` and appears to be made for this type of situation.